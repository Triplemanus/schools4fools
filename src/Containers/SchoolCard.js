import React from 'react';
import './SchoolCard.scss';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const SchoolCard = ({
  school_id,
  school_name,
  phone,
  address,
  distance,
  low_Grade,
  high_Grade,
  level,
  is_Charter,
  is_Magnet,
  is_Private
}) => {
console.log('SchoolCard has been summoned!', school_name, phone, is_Charter, address);
  return (
    <div className="Card">
      <h2>Name: {school_name}</h2>
      <h3>Phone: {phone}</h3>
      <h3>Distance: {distance}</h3>
      {/* <p>Level: {level}</p>
      <p>Lowest Grade: {low_Grade}</p>
      <p>Highest Grade: {high_Grade}</p>
      <p>Charter: {is_Charter}   Magnet: {is_Magnet}   Private: {is_Private}</p> */}
      <Link to={`/schools/${school_id}`}>
        <button className="detail-button" type="button">School Details</button>
      </Link>
    </div>
  )
}

export const mapStateToProps = state => ({
  ...state
});

export default connect(mapStateToProps)(SchoolCard);

SchoolCard.propTypes = {
  school_id: PropTypes.string.isRequired,
  school_name: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  distance: PropTypes.number.isRequired,
  low_Grade: PropTypes.string.isRequired,
  high_Grade: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
  is_Charter: PropTypes.bool,
  is_Magnet: PropTypes.bool,
  is_Private: PropTypes.bool
}