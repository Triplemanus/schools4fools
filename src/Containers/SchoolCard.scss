.Card {
  align-items: center;
  background-color: #F4E697;
  border-radius: 15px;
  border: 4px solid #544061;
  display: flex;
  flex-direction: column;
  height: 350px;
  justify-content: space-evenly;
  margin-bottom: 10px;
  opacity: .8;
  width: 300px;
}

.favoriteCard {
  align-items: center;
  background-color: #F4E697;
  border-radius: 15px;
  border: 4px solid #B54036;
  display: flex;
  flex-direction: column;
  height: 350px;
  justify-content: space-evenly;
  margin-bottom: 10px;
  opacity: .8;
  width: 300px;
}

img {
    height: 150px;
    object-fit: cover;
    width: 150px;
}

h3 {
  align-self: auto;
  padding: 10px;
  text-align: center;
}

button {
  background-color: transparent;
  background-position: center;
  border: none;
}

.favorite {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

#isFavorite, #notFavorite {
  height: auto;
  width: 15%;
}

#isFavorite:hover, #notFavorite:hover, #standardCard:hover {
  border: 2px solid #CC7B3A;
}