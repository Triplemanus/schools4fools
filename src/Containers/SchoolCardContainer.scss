.CardContainer {
  background-color: #384150;
  display: flex;
  flex-flow: row wrap;
  height: 100%;
  justify-content: space-evenly;
  width: 100%;
}

#error {
  color: red;
  font-size: 15px;
  height: 25px;
  text-align: center;
}