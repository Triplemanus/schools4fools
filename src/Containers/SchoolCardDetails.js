import React from 'react';
import { Link } from 'react-router-dom';
import './SchoolCardDetails.scss';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const SchoolCardDetails = ({ schools, error, returnRoute}) => {


// export const SchoolCardDetails = ({  school_name, address, phone, distance, returnRoute }) => {
  console.log('S_C_D state, schoolId is : ', schools);
  //<Link to={`${returnRoute}`} className='back-btn'>◀ back</Link>
  return (
    <section className="CardDetails">
      <Link to="/schools" className='back-btn'>◀ back</Link>
      <article id="cardDetails-container">
      <h1 id="school-name">School Name:  {schools.schoolList[1].schoolName}</h1>
      <h2 id="school-address">School Address:  {schools.schoolList[1].address.street}</h2>
      <p className='card-phone'>School Phone:  {schools.schoolList[1].phone}</p>
      <p className='card-distance'>Low Grade:  {schools.schoolList[1].lowGrade}</p>
      </article>
    </section>
  )
}

export const mapStateToProps = state => ({
  ...state
});

export default connect(mapStateToProps)(SchoolCardDetails);

// SchoolCardDetails.propTypes = {
//   school_name: PropTypes.string.isRequired,
//   address: PropTypes.string.isRequired,
//   phone: PropTypes.string.isRequired,
//   distance: PropTypes.number,
//   returnRoute: PropTypes.string.isRequired
// }