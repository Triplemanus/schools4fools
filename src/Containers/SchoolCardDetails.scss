#cardDetails-container {
  display: flex;
  flex-flow: row wrap;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: rgb(188, 186, 189);
}

#school-name {
  font-size: 20px;
}

.CardDetails {
  align-items: center;
  background-color: rgb(196, 198, 199);
  border-radius: 12px;
  border: 4px solid #544061;
  color: black;
  display: flex;
  flex-direction: column;
  font-size: 1.5vh;
  height: 610px;
  justify-content: space-evenly;
  margin-bottom: 2px;
  opacity: .7;
  width: 470px;
}

.card-low-grade {
  color: black;
  font-size: 14;
}

.card-hi-grade {
  color: black;
  font-size: 14;
}

#school-name {
  color: black;
  font-size: 16;
}

#school-address {
  color: black;
  font-size: 14;
}

#school-address {
  color: black;
  font-size: 14;
}

.back-btn {
  font-size: 2.5em;
  border: none;
  width: 200px;
  border-radius: 5px;
  &:hover {
    color: white;
    background-color: #BC8F8F;
  }
}