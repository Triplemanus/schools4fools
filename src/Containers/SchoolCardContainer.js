import React from 'react';
import SchoolCard from './SchoolCard';
import './SchoolCardContainer.scss'
import { connect } from 'react-redux';

export const SchoolCardContainer = ({ schools, error }) => {

  let testSchool = {
    school_id: "080336006756",
    key: "080336006756",
    school_name: "George Washington High School",
    phone: '303.399.4532',
    address: '1234 S Monaco St.',
    distance: 5.8,
    low_Grade: '9',
    high_Grade: '12',
    level: 'High',
    is_Charter: false,
    is_Magnet: false,
    is_Private: false
  }

let schoolCards;
schools ? ( schoolCards = schools.schoolList.map(school => {
  return (
      <SchoolCard
      school_id={school.schoolid}
      key={school.schoolid}
      school_name={school.schoolName}
      phone={school.phone}
      address={school.address}
      distance={school.distance}
      low_Grade={school.lowGrade}
      high_Grade={school.highGrade}
      level={school.level}
      is_Charter={school.isCharterSchool}
      is_Magnet={school.isMagnetSchool}
      is_Private={school.isPrivate}
      />
  );
})) : ( schoolCards = <SchoolCard {...testSchool}/> )
  return (
    <section>
      {schoolCards}
    </section>
  )
};

export const mapStateToProps = state => ({
  ...state
});

export default connect(mapStateToProps)(SchoolCardContainer);