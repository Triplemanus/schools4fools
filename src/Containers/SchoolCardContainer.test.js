import React from 'react';
import { shallow } from 'enzyme';
import { SchoolCardContainer, mapStateToProps } from '../Containers/SchoolCardContainer';

describe("CardContainer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SchoolCardContainer
        schools={[]}
        error=""
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("mapStateToProps", () => {
  it("should return an object with the contents of the store", () => {
    const mockStore = {
      error: "",
      schools: [
        {
          school_id: "064215006903",
          schoolName: "Helen Stacey Middle",
          phone: "(714) 894-7212",
          address: {
            "latLong": {
              "latitude": 33.747461,
              "longitude": -118.01877
            },
            "street": "6311 Larchwood Dr.",
            "city": "Huntington Beach",
            "state": "CA",
            "stateFull": "California",
            "zip": "92647",
            "zip4": "2320",
            "cityURL": "https://www.schooldigger.com/go/CA/city/Huntington+Beach/search.aspx",
            "zipURL": "https://www.schooldigger.com/go/CA/zip/92647/search.aspx",
            "html": "6311 Larchwood Dr.<br />Huntington Beach, CA 92647-2320"
          },
          distance: 20
        }
      ]
    };

    const expectedState = {
      error: "",
      schools: [
        {
          school_id: "064215006903",
          schoolName: "Helen Stacey Middle",
          phone: "(714) 894-7212",
          address: {
            "latLong": {
              "latitude": 33.747461,
              "longitude": -118.01877
            },
            "street": "6311 Larchwood Dr.",
            "city": "Huntington Beach",
            "state": "CA",
            "stateFull": "California",
            "zip": "92647",
            "zip4": "2320",
            "cityURL": "https://www.schooldigger.com/go/CA/city/Huntington+Beach/search.aspx",
            "zipURL": "https://www.schooldigger.com/go/CA/zip/92647/search.aspx",
            "html": "6311 Larchwood Dr.<br />Huntington Beach, CA 92647-2320"
          },
          distance: 20
        }
      ]
    };

    const testProps = mapStateToProps(mockStore);

    expect(testProps).toEqual(expectedState);
  });
});