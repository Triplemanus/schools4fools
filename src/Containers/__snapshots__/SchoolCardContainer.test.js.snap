// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
<section>
  <Connect(SchoolCard)
    address="1234 S Monaco St."
    distance={5.8}
    high_Grade="12"
    is_Charter={false}
    is_Magnet={false}
    is_Private={false}
    key="080336006756"
    level="High"
    low_Grade="9"
    phone="303.399.4532"
    school_id="080336006756"
    school_name="George Washington High School"
  />
</section>
`;
